lobby/src/main/java/games/strategy/engine/lobby/server/LobbyRunner.java

```diff
@@ -18,16 +19,17 @@ private LobbyRunner() {}
    * Entry point for running a new lobby server. The lobby server runs until the process is killed or the lobby server
    * is shut down via administrative command.
    */
-  public static void main(final String[] args) {
-    try {
-      final LobbyPropertyReader lobbyPropertyReader =
-          new LobbyPropertyReader(new FilePropertyReader("config/lobby/lobby.properties"));
-      log.info("Starting lobby on port " + lobbyPropertyReader.getPort());
-      LobbyServer.start(lobbyPropertyReader);
-      log.info("Lobby started");
-    } catch (final Exception e) {
-      log.log(Level.SEVERE, "Failed to start lobby", e);
-      ExitStatus.FAILURE.exit();
+  public static void main(final String[] args) throws IOException {
+    File propertyFile = new File("config/lobby/lobby.properties");
+    if (!propertyFile.exists()) {
+      log.log(Level.SEVERE, "Could not find property file: " + propertyFile.getAbsolutePath());
+      return;
```

Comment:
The `return` regresses #3679.  If the lobby server fails to start, we should exit with a nonzero status.

Additionally, similar to what we were discussing in #3773, this is effectively a TOCTOU.  Since we don't attempt to recover from an exception thrown at the TOU and let it terminate the process, I'm not sure the TOC buys us anything.  The exception thrown at TOU will contain sufficient information to describe the cause as a missing file, including its path.

Reply:
#3679 was updating from a non-standard exit code, -1, to a more standard one. It is not necessarily standard for java apps to follow unix exit code conventions (notably java is run on many platforms). 

We do not have a requirement or a need to distinguish exit code of the lobby runner process, we do not do anything with it.

The TOC buys only a bit more clarity, instead of tracing it down to see we'll get an exception (control flow via exception handling), the control flow is instead explicit.

Reply:
a2888fe462508a136a2ec630ee37f765b3ea7cd4 updates this, uses a precondition check here - it looks cleaner/simpler IMO that way. We go back to getting an exception. IMO that is okay, prop file missing means a deployment/installation or OS/hardware problem, nothing good.

We still get earlier checking, the code that owns the file path is now also responsible for checking it before it sends a file path to another object. So we are moving the point of failure ahead slightly, that is another reason to add an additional TOC check, to move that up in processing (and hopefully make the control flow a bit easier to follow).

Reply:
> #3679 was updating from a non-standard exit code, -1, to a more standard one.

Actually, #3679 changed the exit status for `LobbyRunner` from `0` to `1` if an exception was thrown while starting the lobby.  Hence, why I indicated it was a regression from that PR to change the exit status back to `0` upon failure.  Changing an exit status of `-1` to `1` throughout the codebase was done in #3690.

Regardless, my original comment is moot now that an exception is being raised.  The Oracle JRE, at least, sets the exit status to `1` if an unhandled exception terminates the application.